### DISCUSSION TITLE
Distance metric for comparing embeddings

### DISCUSSION THREAD

**Discussion by kentwait** [facebookresearch/esm#485] [General]
In Rives et al, 2021 paper in PNAS, cosine similarity was used to compare site- and sequence-based embeddings generated by ESM1b to show that these latent representations capture biologically meaningful properties such as aligned/orthologous sites and secondary structure.

What was the reason for using cosine similarity over other common distance metrics such as dot-product, L1 (Manhattan) or L2 (Euclidean)?

**Comment by tony-res** [comment]
Cosine similarity goes from -1 to +1. +1 means the two embeddings are exactly alike, i.e. point in the same direction. 0 means the two embeddings are orthogonal (point in directions that are 90 degrees apart). -1 means the two embeddings are pointing in completely opposite directions. It's a little easier to think about a direction that goes between -1 and +1 rather than L1/L2/etc which are real numbers and can go from 0 to infinity. If two vectors have an L2 of 10.3 does that mean they are "close"? It's a little easier to think in terms of -1 to +1.
